### DISCUSSION TITLE
Unterstützung für weiteren Stromerzeuger?

### DISCUSSION THREAD

**Discussion by niceguy0815** [solectrus/solectrus#2621] [General]
Hi,
ich habe eine Senec Anlage auf meinem Dach und diese läuft auch schon eine ganze Weile in solectrus.
Jetzt habe ich vor mir eine zweite Angage ohne Speicher auf Garage und Schuppen zu bauen (Weil zwei Specher im selben Netzt ist galube ich schwer umzusetzen?) .
Und ich farge mich ob es möglich ist diese auch z.B mit mqtt auch in die Selbe solectrus installation zu bekommen?
Oder benötige ich für die zweite Anlage mit Mqtte auch eine komplette neue extra solectrus installation.

Des weiteren frage ich mich (hat aber nix mit solectrus zu tun) ob der erzeugte Strom der zweiten Anlage auch in die Senec Home v 3 hybrid eingespeißt wird oder ob nur das in den Akku fließt was über den Senec Wechselrichter kommt?
Eventuell weiß das jemand?

Dachte ich farg hier einfach mal, vieleicht hat ja jemand eine Idee?

**Comment by ledermann** [comment]
Ob und wie ein solches Szenario elektrotechnisch umsetzbar ist, vermag ich nicht zu beurteilen.

Bezüglich SOLECTRUS könntest Du mit anderen Tools die Daten zusammenführen, also insbesondere die Summe der Erzeugung bilden. Diese Daten könnten dann per MQTT verteilt und vom MQTT-Collector empfangen werden. Der SENEC-Collector käme dann gar nicht zum Einsatz.

Insgesamt klingt das aber nach einem wilden Ritt :-)

**Comment by Zeppelin-PV** [comment]
Solch ein Szenario ist durchaus praktikabel
Eine Senec Anlage speißt auf zwei Phasen den erzeugten Strom ins Netz (L1 und L2 in meinem Fall)
Ein BKW kann somit auf der dritten Phase einspeißen ohne den Senec zu stören.
Ich habe mein BKW angemeldet und mir wurde vorgeschlagen dies als Erweiterung der vorhandenen PV-Anlage zu tun.
Somit komme ich auch in den Genuss einer Vergütung bei Einspeißung.
Aber auch ich hätte gerne dass das BKW in Solectrus auf der Erzeugungsseite angezeigt wird.
Ich habe einen Shelly installiert der dazu zur Verfügung stehen würde.
Dies habe ich auch aktuell im Issue Wärmepumpe gepostet.

**Reply by ledermann** [reply]
Da ich in #2239 gerade die Darstellung separater Verbraucher bearbeite, tun sich hier tatsächlich Parallelen auf. Ich habe aber Verständnisfragen:

Mal angenommen, Dein Shelly beobachtet L3 und schiebt die gemessene Erzeugung des BKW in die InfluxDB. Wie wäre das dann in SOLECTRUS darzustellen? Meine Vermutung:

1. Links in der Strombilanz (Herkunft) wird der L3-Messwert als zusätzlicher Erzeuger angezeigt, also zusätzlich zur bereits bestehenden Erzeugung vom Dach.
2. Damit die Bilanz ausgeglichen wird, muss der Wert auch rechts (Verwendung) für eine Erhöhung sorgen. Aber wie? Erhöhung des Hausverbrauchs um diesen Wert? Oder Erhöhung der Einspeisung?

Mit dem kürzlich erstellten [Shelly-Collector](https://github.com/solectrus/shelly-collector) kannst Du ja schon mal experimentieren. Die derzeitige Implementierung erfasst die drei Phasen (Active Power A/B/C) und den Gesamtwert (Active Power Total).

Grundsätzlich könnte ich mir vorstellen, SOLECTRUS so zu erweitern, dass separate Verbraucher und Erzeuger zusätzlich dargestellt werden können. Bei den Verbrauchern scheint mir die Rechnung einfach zu sein: Die werden vom bestehenden Hausverbrauch abgezogen. Wie bei zusätzlichen Erzeugern gerechnet werden soll, wäre noch zu klären.

**Comment by Zeppelin-PV** [comment]
Der Ertrag des BKW müsste links bei der Erzeugung dargestellt werden, als eigenständiger Graph mit einem anderen Grünton und einem entsprechenden Symbol für das BKW. die Fuktion ist folgende: wenn das BKW ordentlich erzeugt geht zunächst der Hausverbrauch zurück und zwar soweit bis NULL, dann steigt der Betrag bei der Einspeisung über den der links vom Dach dargestellt wird.
Rechnerisch wäre das dann so wie es in diesem Screenshot zu sehen ist.
Eine andere Beobachtung ist folgende: wie in dem Bild zu sehen ist wird in der Grafik Hausverbrauch immer ein Peak größer 4kW kurzzeitig angezeigt und ist dann beim nächsten Refresh wieder weg, aber deshalb kann die Skalierung nicht angepasst werden.
Wenn es dazu eine Deiner guten Beschreibungen geben würde, wäre super

![grafik](https://github.com/solectrus/solectrus/assets/126392805/ae488850-8b6a-48de-9220-54ae98ef107b)

**Reply by ledermann** [reply]
Ok, danke für die Erläuterungen. Ich bin allerdings weiter unsicher, wie dann gerechnet werden soll.

Nehmen wir mal an, in einem bestimmten Moment meldet dein Shelly eine BKW-Erzeugung von `300 W`, während dein SENEC meldet, dass vom Dach `3 kW` kommen. Die `300 W` packe ich dann links auf die `3 KW` drauf, man würde also zwei Segmente in unterschiedlichen Grüntönen sehen, `3 kW + 300 W`. Ok, das war einfach.

Auf der rechten Seite ist es kniffliger. Nehmen wir an, dein SENEC meldet einen Hausverbrauch von `0 W` und eine Einspeisung von `3,1 kW`. Dann rechne ich jetzt rückwärts den tatsächlichen Hausverbrauch: `3,3 kW - 3,1 kW = 200 W`. Damit also die Bilanz stimmt, muss der tatsächliche Hausverbrauch `200 W` betragen und auch so angezeigt werden (anstelle der `0 W`, die von SENEC gemeldet wird). Die Idee ist es also, den Hausverbrauch rechnerisch so weit zu erhöhen, dass die Bilanz ausgeglichen ist.

Stimmt das so?

Für deine Frage zum Peak habe ich leider keine Lösung parat, die Skalierung ergibt sich immer aus Min und Max. Das ist aber auch ein separates Thema.

**Reply by bennysweb** [reply]
Ja, so würde ich das Ganze auch verstehen. Habe den gleichen Use-Case und wäre ebenfalls sehr an der Umsetzung interessiert:-)

**Comment by Zeppelin-PV** [comment]
... natürlich könnte man dann das BKW mit in die Forecast hinzufügen, z.B. 820Wp, 180°, 30°Azimuth

**Comment by Zeppelin-PV** [comment]
Ich habe nun viele Tage betrachtet in dem das BKW den Verbrauch dezimiert, der dann u.U. auf NULL geht, und es somit zu einer Ungleicheit zwischen Erzeugung und Verbrauch kommt. Dann aber ist es in der Tat so, dass mehr eingespeißt wird als verbraucht.
Ich zeige mal die Screenshot vom 13.3.24 um 13:20h von Solectrus, vom Wechselrichter BKW und von der Senec App.
Der Senec Graph kreiert eine weiße Fläche die den Überschuss des BKW darstellt. Wenn Solectrus das BKW anzeigen würde müsste es ja im Prinzip das selbe passieren. Rechts bei Einspeißung könnte man dann ab dem Moment wo die Parität überschritten wird ein Einspeiße Graph anzeigen der den Überschuss darstellt. Grau "Hausverbrauch" ist NULL und wird nicht mehr angezeigt. Dafür ein zusätzlicher Graph BKW Einspeißung. Mit dieser Darstellung wäre quasi die Anzeigeparität wieder hergestellt.

![grafik](https://github.com/solectrus/solectrus/assets/126392805/c7fa938a-cc04-4f1f-954c-a9ad1b98c82c)

![grafik](https://github.com/solectrus/solectrus/assets/126392805/c523e63a-5c4f-429e-a44e-aec8bd7fc9e4)

![grafik](https://github.com/solectrus/solectrus/assets/126392805/0d43e84a-792d-4280-8909-5ac3e0e15e16)

**Comment by ledermann** [comment]
Ich tue mich noch etwas schwer damit, wie nun wirklich gerechnet werden soll. Insbesondere die rechte Seite ist unklar. Lassen wir mal die Graphen zunächst beiseite und nur die Tagessummen nehmen. Ich entnehme den Screenshots von @Zeppelin-PV vom 13.2.2024 folgende Werte:

| Herkunft              | Verwendung         |
| --------------------- | ------------------ |
| Netzbezug 6,7         | Hausverbrauch 10,1 |
| Erzeugung (Dach) 22,6 | Einspeisung 19,3   |

Nun wird vom BKW eine Erzeugung von 2,2 gemeldet. Links wird das als zusätzliches Segment angezeigt. Wo bringe ich diese Energie aber rechts unter? Meiner Ansicht nach muss sie **ausschließlich** dem Hausverbrauch zugerechnet werden. Es findet ja keine zusätzliche Einspeisung statt und es gibt auch sonst keine Verwendungsmöglichkeit.

Das Endresultat wäre also das hier:

| Herkunft              | Verwendung         |
| --------------------- | ------------------ |
| Netzbezug 6,7         | Hausverbrauch 12,3 |
| Erzeugung (Dach) 22,6 | Einspeisung 19,3   |
| Erzeugung (BKW) 2,2   |                    |

Einverstanden?

**Reply by megaholti** [reply]
ich habe im letzten Post mal mein Formelgedankengang hinterlegt.

**Comment by bennysweb** [comment]
Ich glaube, so einfach ist es dann doch nicht. Das BKW ist wahrscheinlich auf die 3. Phase geklemmt um eine Drosselung des Senec-Wechselrichters auf Grund von Schieflast auf einer Phase zu vermeiden - Senec ist ja nur zweiphasig ans Hausnetz angeschlossen. Somit kann es unter Umständen doch sein, dass das BKW über die dritte Phase einspeist. Das macht die Anzeige auf der rechten Seite leider nicht einfacher, das ist mir bewusst. Ich habe ehrlich gesagt auch keinen schlauen Vorschlag für das Problem.

**Reply by ledermann** [reply]
Ok, das hatte @Zeppelin-PV oben bereits angedeutet, was ich aber nicht ganz verstanden habe. Wenn also sowohl der SENEC-Wechselrichter einspeisen kann als auch das BKW, dann fehlt mir in der ganzen Darstellung noch eine Institution, die die Gesamteinspeisung ermittelt. Zumindest einen Zähler des Netzbetreibers wird es ja geben. Dessen Messwerte könnten helfen.

**Comment by bennysweb** [comment]
Die Gesamteinspeisung wird über den EnFluRi zurück an Senec gemeldet, ist also SOLECTRUS bekannt. Wenn die Sonne gut steht, übersteigt die Einspeisung rechts die Erzeugung links und der Hausverbrauch sinkt auf 0. Das müssen wir irgendwie rechnerisch auseinander dröseln. Am 27.02.2024 um 18:49 schrieb Georg Ledermann ***@***.***>:﻿
Ok, das hatte @Zeppelin-PV oben bereits angedeutet, was ich aber nicht ganz verstanden habe. Wenn also sowohl der SENEC-Wechselrichter einspeisen kann als auch das BKW, dann fehlt mir in der ganzen Darstellung noch eine Institution, die die Gesamteinspeisung ermittelt. Zumindest einen Zähler des Netzbetreibers wird es ja geben. Dessen Messwerte könnten helfen.

**Comment by Zeppelin-PV** [comment]
Ich habe mir mal die Zeit genommen und den EnFluRi im Zählerkasten beobachtet.
Dieser zeigt mir die Leistung aller drei Phasen. Wird Strom generiert, egal ob Dach oder BKW, gehen die positiven Werte nach unten und,
wenn der Wert größer wird als die Last an einer Phase, wird ein negativer Wert angezeigt, sprich es wird eingespeißt.
Dieses Verhalten spiegelt sich im Prinzip bei Solectrus wieder. 
Der SENEC speißt auf zwei Phasen ein, verrechnet aber die Last aller drei Phasen. Selbst wenn L1 und L2 ins minus gehen und hier eingespeißt wird, wird trotzdem noch eine Last sprich Hausverbrauch angezeigt. Dieser stammt dann von L3. Das BKW das an L3 arbeitet drückt zunächst den Hausverbrauch nach unten. Bei ausreichender Erzeugung soweit bis der Hausverbrauch null ist. Jetzt speißt sogar das BKW ein.
Auf Grund der momentanen Sonneneinstrahlung sind es nur relativ kurze Phasen wo dies zu beobachten ist. Wenn in ein paar Wochen das BKW mal über 2 Stunden 700W oder mehr erzeugt wird die Situation vielleicht besser darzustellen sein.
Zum Darstellungsprinzip in Solectrus das ja auf Parität setzt zwischen Erzeugung und Verbrauch wird es schwierig.
Vielleicht am Einfachsten wenn die Produktion des BKW links und Einspeisung rechts gleich angezeigt wird.
Geht der Hausverbrauch zurück auf Grund hoher Einspeisung des BKW steigt der Wert Einspeisung BKW
In der Situation Hausverbrauch null, bekommt das BKW den Überschuss on top. So wäre die ganze Zeit Erzeugung und Verbrauch gleich.

**Comment by Zeppelin-PV** [comment]
habe mal versucht einen Tag darzustellen mit konstantem Hausverbrauch
die Werte sind rein theoretisch, aber zeigen u.U. wie sie in Solectrus verrechnet werden müssten.

![grafik](https://github.com/solectrus/solectrus/assets/126392805/01d6cb44-422d-49a6-b23c-c2049076f5cf)

**Reply by Zeppelin-PV** [reply]
da ist in der Tabelle ein Fehler drin
hier nochmal

![grafik](https://github.com/solectrus/solectrus/assets/126392805/75972cb3-5695-4829-b8b7-592ecceb9aca)

**Reply by Zeppelin-PV** [reply]
das BKW speist eigentlich immer ein, zum Einen ins Hausnetz an die Verbraucher, zum Anderen, wenn diese "gesättigt" sind, ins Netz.
Der Balken rechts müsste heißen "Reduktion Hausverbrauch-Einspeisung"

**Comment by ledermann** [comment]
Danke Euch für die vielfältigen Hinweise, aber ich bekomme es nicht hin, daraus einfache Formeln zur Berechnung von Einspeisung und Hausverbrauch abzuleiten. Da ich selber kein Balkonkraftwerk habe, kann ich auch keine eigenen Experimente durchführen.

Zurzeit spielt die Musik in #2239, ich lasse dieses Thema daher erst mal ruhen. Wenn es neue Erkenntnisse gibt oder mir jemand aufzeigen, wie ich das in SOLECTRUS berechnen soll (Formeln!), dann komme ich gerne wieder darauf zurück.

**Reply by bennysweb** [reply]
Ich hätte folgenden Vorschlag: 
Der Senec-Collector muss die aktuelle Netz-Leistung der dritten Phase abfragen (z.B. Homeassistant kann das - müsste also gehen). Ist dieser Wert positiv, kann die Erzeugung des BKW vom Hausverbrauch abgezogen werden. Ist er negativ, ist das genau der Wert, den das BKW einspeist und ist somit der Einspeisung zuzuordnen.

Voraussetzung ist natürlich, dass das BKW an die dritte Phase geklemmt ist. Aber nur das macht auch technisch am meisten Sinn, da wie schon geschrieben Senec selbst nur zweiphasig ans Hausnetz angeschlossen ist. Wenn darauf noch ein zusätzliches BKW angeklemmt wäre, würde der Wechselrichter des Senec herunterregeln um eine Schieflast zu vermeiden.

Wäre das denkbar?

**Reply by StevieC121176** [reply]
Also mein Usecase sieht da etwas anders und einfacher aus. Ich habe mein Balkonkraftwerk schon in Solectrus über den MQTT Collector eingebunden. Man kann ja schon 3 MPP Stränge mitloggen und einmal die Gesamterzeugung. Einzig die Anzeige der einzelnen Stränge in Solectrus fehlt noch. Dann könnte man Balkonkraftwerke über diesen Weg sichtbar machen. Bei mir ist jeder Strang ein Wechselrichter. 

sh auch hier: https://github.com/solectrus/solectrus/issues/1756#issuecomment-1674588610

**Comment by Zeppelin-PV** [comment]
möchte mich nochmals zu der Darstellung eines BKWs zur vorhandenen PV-Anlage melden.
Das Verständnis dass zwischen Herkunft und Verwendung Parität herrschen soll ist klar.,
Das BKW ist ein Erzeuger ohne Intelligenz, sag ich mal so; während eine PV-Anlage austariert wie der Strom verwendet werden soll; Haus, Akku, Netz. Das BKW kann das nicht, der Strom wird einfach verbraten. In meinem Fall alles was auf L3 angeschlossen ist.
Zur Darstellung in Solectrus wäre es dann in der Tat einfach.
Wird vom BKW Strom erzeugt erscheint in Solectrus in Herkunft ein etwas anders gefärbter grüner Balken mit dem Wert (Watt) und einem Sonnensymbol mit BKW als Beschriftung.
Wenn in der Solectrusberechnung nichts geändert werden soll, wird unter Verwendung das BKW als Einspeiser angezeigt, vielleicht mit einem Doppelsymbol Haus und Netz. Die Anzeige ist identisch mit der Erzeugung, die Parität bleibt gewahrt.
Eine Art Überwachungsrechnung könnte folgendermaßen aussehen:
Summe Herkunft = Summe Verwendung ohne BKW; alles was in dieser Berechnung darüber liegt wird dem Erzeuger BKW zugerechnet.
Man könnte noch mit den zwei Symbolen spielen. Speist das BKW ins Haus ein und schmälert den Hausverbrauch wird ein Haus angezeigt, ist der Hausverbrauch Null wird z.B. ein Strommast angezeigt.
... hoffe das kommt einigermaßen verständlich rüber

**Reply by Zeppelin-PV** [reply]
zur Veranschaulichung gebastelt   ... aber so in etwa könnte ich mir das vorstellen, nur schöner  ;-)

![grafik](https://github.com/solectrus/solectrus/assets/126392805/d73416db-300e-42ca-a3dd-9a7611cc3d9e)

**Comment by ledermann** [comment]
Vielen Dank für den zusätzlichen Input. Ich werde da gerne drauf zurückkommen. Momentan konzentriere ich mich auf die Fertigstellung der Unterstützung von Wärmepumpe und externer Wallbox. Hier gibt es noch einiges zu tun, bevor ich das als Version 0.15 veröffentlichen kann. Unter anderem gestalte ich die Konfiguration flexibler, um überhaupt weitere Geräte anbinden zu können. Noch ist SOLECTRUS intern zu sehr an SENEC orientiert, zukünftig wird der MQTT-Collector an Bedeutung gewinnen.

Unterstützung für zusätzliche Balkonkraftwerke (und Wallboxen) kommt somit später. Daher bitte noch etwas Geduld :)

**Comment by StevieC121176** [comment]
Ich würde das nicht so darstellen. 
Ich habe auch ein einphasiges Balkonkraftwerk zusätzlich zu meinen beiden Kostal WR.
Am Energiemeter wird die Einspeisung, Verbrauch und Netzbezug gemessen. Aber saldiert von allen Phasen. Man weiss also nie ob die Erzeugung des einzelnen WR und BKW gerade für den Verbrauch genutzt oder eingespeist wird. Einzig die Erzeugerleistung kann von jedem Gerät abgefragt werden. 
Daher würde ich auch nur die Erzeugerseite aufsplittern und die Verbrauchsseite so lassen wie sie ist.

Als Beispiel hatte ich diese Darstellung mal aufgeführt.

![259992866-d7641589-abe5-4d10-bccd-5de3787d8479](https://github.com/solectrus/solectrus/assets/113791299/16d24135-edd9-4e54-914f-c77c0dc7600f)

**Comment by megaholti** [comment]
Ich hätte da eine Idee, es mathematisch umzusetzen:
Wir haben ja folgenden Fakt:  auf der Erzeugerseite a+b+c (a=Senec-PV, b=BKW-PV und c=Netzbezug) = Verbrauch x+d+e (x=Hausverbrauch, d=Beladung und e=Einspeisung). Da alle Werte realistisch vorliegen nur x der Hausverbrauch nicht korrekt vom Senec wiedergegeben wird, wäre die Formel:

X=(a+b+c)-(d+e) also Hausverbrauch=(Senec-PV+BKW-PV+Netzbezug)-(Beladung+Einspeisung).

Was denkt Ihr?

![IMG_2214](https://github.com/user-attachments/assets/42b2ffca-23c1-45dc-98a9-17bd2333a054)
![IMG_2211](https://github.com/user-attachments/assets/af48ac76-c9bf-4593-91f4-7f671d4a3e77)

**Comment by ledermann** [comment]
Aufgrund des Feedbacks von @megaholti (wir haben das noch per E-Mail diskutiert) scheint es nun etwas einfacher auszusehen. Ich fasse kurz zusammen:

1) Der Hausverbrauch, den SENEC meldet, ist bei gleichzeitiger Verwendung einer zusätzlichen PV-Quelle falsch und kann nicht benutzt werden
2) Da die Strombilanz immer ausgeglichen sein muss (Herkunft = Verwendung) kann aber ein korrigierter Hausverbrauch berechnet werden

Für die beiden Seiten der Strombilanz muss gelten:

```
Herkunft = PV_SENEC + PV_BKW + Netzbezug + Akkuentnahme
Verwendung = Hausverbrauch + Akkubeladung + Einspeisung (+ E_Auto + Wärmepumpe)
```

Geht man davon aus, dass genau ein Wert (z.B. der Hausverbrauch) unbekannt oder falsch ist, kann man diesen rückwärts aus der Strombilanz berechnen:

```
Hausverbrauch = Herkunft - Akkubeladung - Einspeisung (- E-Auto - Wärmepumpe)
```

Können das mal diejenigen überprüfen, die bereits ein BKW am Start haben?

Wenn das so stimmt, könnte die Implementierung so aussehen:

1) Zusätzlichen Sensor für BKW-Erzeugung ermöglichen
2) Sonderbehandlung, wenn Sensor für Hausverbrauch leer ist: Der wird dann berechnet über obige Formel

Als User mit einem BKW würde man also einen Sensor hinzufügen und den Hausverbrauch leer lassen. Das klingt einfach.

**Reply by Zeppelin-PV** [reply]
Ich bin mir immer noch nicht ganz schlüssig ob meine Beobachtungen letztes Frühjahr in dieser Formel gerechter werden.
Zumindest konnte ich konkret beobachten, dass die BKW Produktion sich direkt auf die Minimierung des Hausverbrauchs auswirkt.
siehe meine Bilder weiter oben vom 21. und 23. Feb. 2024

Herkunft = PV_SENEC + PV_BKW + Netzbezug + Akkuentnahme
Verwendung = (Hausverbrauch-PV_BKW)* + Akkubeladung + (Einspeisung+PV_BKW)*
* wenn die Produktion des BKW so hoch ist, dass der Hausverbrauch 0 wird, dann wird der Restbetrag der BKW Produktion der Einspeisung giutgeschrieben, sprich hinzu addiert

visuell müsste es derart gestaltet werden, dass es sichtbar ist dass die BKW Produktion zuerst den Hausverbrauch schrumpft (kompensiert)*
bzw. bei dessen 0 die Einspeisung um den Restbetrag des BKW ansteigt
*dies ist ja gewissermaßen auch ein Verbrauch wenn wenn die Energie des BKW dazu benutzt wird (Kompensation) um den Hausverbrauch zu dezimieren

vielleicht ist dies rechnerisch schwieriger umzusetzen als
Hausverbrauch = Herkunft - Akkubeladung - Einspeisung (- E-Auto - Wärmepumpe)
... keine Ahnung

**Reply by megaholti** [reply]
Genau das passiert ja auch, ist bei mir auch so, darum kann der Wert Hausverbrauch aus dem SENEC Speicher nicht genutzt werden und müsste errechnet werden.

**Reply by Zeppelin-PV** [reply]
der SENEC zeigt doch das an was ihm der EnFluri ins Ohr flüstert, d.h. der angezeigte Wert Hausverbrauch müsste doch den Tatsachen entsprechen. Wenn ich nun an L3 ein BKW betreibe (SENEC an L1 und L2) bekommt das der Enfluri mit und der SENEC rechnet das dem Hausverbrauch zu bzw. zieht das ab. Man müsste das mal anhand nur der EnFluri Daten rechnen.
Jetzt, momentan, sieht es natürlich traurig aus
![grafik](https://github.com/user-attachments/assets/418d7125-508a-4b45-9b60-a4328a99c7fd)
![grafik](https://github.com/user-attachments/assets/91b84d4f-1f36-4af4-aaad-8fb4a3229c9a)

**Comment by megaholti** [comment]
Genau das macht er eben nicht - wie auf meinen Screenshots zu sehen, aber bei uns ist heute sonnig, meine aktuellen Werte: PV Senec 3,9 + PV Zaun (4xBKW)437 Watt - Hausverbrauch 266w (Stimmt nicht)  1,1kw Einspeisung, 2,5kw Beladung 

Der Hausverbrauch müsste 703w sein, also 437+266

**Reply by megaholti** [reply]
<img width="2048" alt="Bildschirmfoto 2025-01-08 um 09 59 03" src="https://github.com/user-attachments/assets/5d242838-3ba5-4d5c-9e28-7b32e701479b" />
<img width="2048" alt="Bildschirmfoto 2025-01-08 um 09 58 12" src="https://github.com/user-attachments/assets/f61611c3-4902-4ece-ae22-7fb651a9d7a7" />
<img width="2048" alt="Bildschirmfoto 2025-01-08 um 09 57 06" src="https://github.com/user-attachments/assets/523d24f4-8c51-497b-95bb-a2e7dc0b55af" />

**Reply by megaholti** [reply]
Gegen später dreht die Sonne und der Effekt wird deutlicher - mehr Energie auf den BKW und weniger beim Dach Senec. Da schicke ich noch einmal Daten

**Reply by megaholti** [reply]
ich glaube das sagt alles

**Comment by megaholti** [comment]
![IMG_2239](https://github.com/user-attachments/assets/1a204ba0-8ba3-4409-acd9-110d0ad6c966)

**Reply by Zeppelin-PV** [reply]
... es ist wohl mitentscheidend wo die BKWs eingespeißt werden,
demnach gebe ich jetzt an dieser Stelle klein bei und gehe davon aus dass Georg die richtige, praktikable Lösung finden und einsetzen wird

**Comment by ledermann** [comment]
Ich habe jetzt mal ein Issue angelegt: solectrus/solectrus#3854

**Reply by StevieC121176** [reply]
Könnte man in diesen Zuge evtl. gleich das Thema mehrere Dachflächen mit aufgreifen? im Endeffekt wäre eine 2. Dachfläche nicht anders zu behandeln als ein BKW?
Mein Usecase ist aktuell 2 Dachflächen + 1 BKW. Zur Zeit addiere ich alle Dachflächen und das BKW zusammen und übergebe es per MQTT an Solectrus. 
Darstellen könnte man das analog der neuen Hausverbrauchsansicht. Anstatt der Verbraucher dann die einzelnen Erzeuger.

**Reply by ledermann** [reply]
Ist die zweite Dachfläche nicht Teil der PV-Anlage auf dem Dach, also `Erzeugung = Dach1 + Dach2`?
Wenn du das BKW bereits jetzt integriert hast, dann stimmt der Hausverbrauch doch nicht, oder?

Ich würde das Feature gerne überschaubar halten. Meiner Erfahrung nach verzettelt man sich leicht, wenn zu viel auf einmal gemacht wird. Insofern würde ich eine separate Ansicht der Erzeuger (dennoch gute Idee!) lieber zurückstellen.

Außerdem: Es hat sich für die BKW-Unterstützung (wie in #3854 dargestellt) bereits ein Sponsor gefunden, die Details werden noch ausgetüftelt.

**Reply by StevieC121176** [reply]
Bei mir wird jede Dachfläche mit einem extra Wechselrichter (um die Verluste durch Verschalungen zu minimieren) betrieben. Sind im Grunde genommen 2 PV Anlagen. Das Balkonkraftwerk ist dann eigentlich die 3. PV Anlage. Ich habe alle Anlagen in EVCC eingebunden und das liefert mir Hausverbrauch, Erzeugung gesamt usw. alles per MQTT, welches ich dann per MQTT Collector an Solectrus übergebe.

**Reply by megaholti** [reply]
ist ja im Prinzip das gleiche, nur, dass es sich nicht um ein Balkonkraftwerk, sondern um eine sep. Anlage handelt. Die Daten kennt Senec nicht, somit auch solectrus nicht. der Vorteil hier wäre nur, dass der Hausverbrauch theoretisch auch vorliegt. Der Einfachheit in der Programmierung von Herrn Ledermann aber auch berechnet wird. Fein, sollte meiner Meinung auch klappen

**Comment by ledermann** [comment]
Ich habe mir die Möglichkeiten einer Implementierung zwischenzeitlich näher angesehen. Das Hinzufügen eines weiteren Sensors (für das BKW) ist unproblematisch. Aber die notwendige Umstellung des Hausverbrauches auf eine berechnete Formel ist deutlich aufwändiger. An recht vielen Stellen geht SOLECTRUS davon aus, dass der Hausverbrauch in einem InfluxDB-Feld steht. Auch der PowerSplitter greift darauf zu. Hinzu kommen die Diagramme für den Hausverbrauch und die Tageszusammenfassungen. Es wäre einiges anzupassen, damit jeweils die Berechnung erfolgt.

Daher folgende Idee, insbesondere an @megaholti: Wäre es machbar, die Berechnung außerhalb von SOLECTRUS vorzunehmen, also z.B. in Home Assistant? Also dass dort der korrigierte Hausverbrauch berechnet und in neues Feld geschrieben wird? Dann hätte SOLECTRUS damit nichts zu tun und würde einfach dieses Feld benutzen.

**Reply by megaholti** [reply]
Das wäre schön, ich habe es probiert, leider ist es meines Wissens nicht so einfach, komplexe Berechnungen durchzuführen. Ich habe versucht für jeden Zwischenschritt einen Rechensensor anzulegen. Das funktioniert gut, bis der Akku geladen wird, dann ist es irgendwo komisch. 

Mir erschließt sich das Rechnen mit den Helfern nicht.

**Reply by max-horvath** [reply]
Ich würde nicht empfehlen, diese Funktionalität von einem externen System abhängig zu machen. Dies würde das System dann komplexer im Aufsetzen und der Pflege/Wartung, vor allem, wenn es Updates gibt und man als Nutzer aufpassen muss, dass alles noch wie erwartet funktioniert. Hier könnte es bei verschiedenen Lösungen ein unterschiedliches Verhalten geben und dann wird es spannend im Verstehen, warum etwas nicht wie erwartet funktioniert. Deshalb würde ich eher befürworten, es direkt in Solectrus zu unterstützen.

**Reply by megaholti** [reply]
Ja dem stimme ich voll uns ganz zu!

**Reply by bennysweb** [reply]
Bei mir rechnet die Wallbox (openWB) den korrekten Hausverbrauch aus und stellt den Wert per MQTT zur Verfügung. Vielleicht können das auch noch andere Wallboxen, dann müsste man nicht mit HomeAssistant herumexperimentieren...

**Reply by StevieC121176** [reply]
Bei mir rechnet EVCC den Hausverbrauch komplett aus und stellt ihn per MQTT Solectrus zur Verfügung. Das Funktioniert, genau wie mit den anderen Werten über MQTT sehr gut.

**Reply by ledermann** [reply]
Danke für die zahlreichen Rückmeldungen. Ich sehe ein, dass eine Abhängigkeit von externen Tools nicht das Wahre ist. Wenn man eine Wallbox oder evcc verwendet, die den korrekten Hausverbrauch auch bei BKW-Verwendung richtig errechnet, dann kann man das nutzen und hat kein weiteres Problem. Aber wer weder ein E-Auto hat noch HomeAssistant bemühen möchte, käme hier nicht weiter.

Insofern muss ich das wohl in SOLECTRUS integrieren. Kurzfristig passt mir das zwar nicht so recht, aber mal sehen, wie sich das bewerkstelligen lässt.
